pkg/virt-controller/watch/vm.go

```diff
@@ -2401,3 +2403,19 @@ func (c *VMController) resolveControllerRef(namespace string, controllerRef *v1.
 	}
 	return vm.(*virtv1.VirtualMachine)
 }
+
+func autoAttachInputDevice(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) {
+	autoAttachInput := vm.Spec.Template.Spec.Domain.Devices.AutoattachInputDevice
+	// Default to False if nil and return
+	if autoAttachInput == nil || (autoAttachInput != nil && *autoAttachInput == false) {
+		return
+	}
+	// Only add the device with an alias here. Preferences for the bus and type might
+	// be applied later and if not the VMI mutation webhook will apply defaults for both.
+	vmi.Spec.Domain.Devices.Inputs = append(
+		vm.Spec.Template.Spec.Domain.Devices.Inputs,
+		virtv1.Input{
+			Name: "default-0",
+		},
+	)
```

Comment:
Do we want to add the input device even if the `vm.Spec.Template.Spec.Domain.Devices.Inputs` is not empty?

Reply:
ACK done, this aligns with the behaviour for the default network interfaces IIRC so this should also be fine here.